### PR TITLE
arch/risc-v: Apply misaligned access handler for k210/bl602

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary
Add a new option to use misaligned memory access for K210/BL602
## Impact
New function for K210/BL602
## Testing
Custom board

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: arch/risc-v/src/common/Kconfig [hunk 1]
```diff
@@ -27,3 +27,8 @@ config RISCV_TOOLCHAIN_GNU_RVGW
 		configured for riscv64-unknown-elf.
 
 endchoice
+
+config RISCV_SOFT_MISALIGNED_HANDLER
```
Comment (xiaoxiang781216) [inline arch/risc-v/src/common/Kconfig:31]: Remove SOFT_?

Comment (no1wudi) [reply arch/risc-v/src/common/Kconfig]: Done

File: arch/risc-v/Kconfig [hunk 2]
```diff
@@ -194,6 +196,17 @@ config ARCH_HAVE_S_MODE
...
+	default n
+	---help---
+	    The chip will raise a exception while misaligned memory access.
+
+config RISCV_MISALIGNED_HANDLER
+	bool "Software misaligned memory access handler"
+	depends on ARCH_HAVE_MISALIGN_EXCEPTION
+	default y
+
 # Option to run NuttX in supervisor mode. This is obviously not usable in
```
[the remark this answers is not in the document]
Comment (no1wudi) [reply arch/risc-v/Kconfig]: OK

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
ci error is fixed by https://github.com/apache/incubator-nuttx/pull/6055, let's merge this PR directly to save the resource.

**Pull request merged**